### PR TITLE
Improve external storages settings ui: Show dedicated messages for failed to load mount points / Autofocus available for while adding external mount points 

### PR DESCRIPTION
**Pull request by AlexAndBear**
## Description
Enhancement: Improve admin external storage settings UI

Before this PR no error notification was shown while an external mount point configuration was not able to load.
This was only indicated with a red square with a long list of external mount points, this was not handy.
Therefore an error notification will be shown with the dedicated external mount point which fails to load.

As well improved the add external mount point functionality with opening the available for select while adding an external mount point.
The select will have now a 'select all' item.
These changes prevents the admin to expose the mount point
unwanted to all users immediately.

## Related Issue

- Fixes https://github.com/owncloud/enterprise/issues/4585

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [x] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by jvillafanez** [comment]
Double-check with multiple mount points. I think that the initial page load triggers an update status for all the mounts, so if there are several wrong mount points the notifications could be piled up.

**Comment by JanAckermann** [comment]
@jvillafanez 
> Double-check with multiple mount points. I think that the initial page load triggers an update status for all the mounts, so if there are several wrong mount points the notifications could be piled up.

That's intended, customer have about 70 mount points with a large list and wants to be informed about all failing mount points without scrolling trough the list

**Comment by jvillafanez** [comment]
My worry is that, if there is a connection problem and all the storages are disconnected, the admin will have 70 notifications in the web UI, and I'm not sure if the UI is ready to handle such amount. It could be fine for 3-5 errors but I think the UI will be bloated if there are more.
I don't remember a scroll in the notification area, so I suspect the area will keep expanding.

**Comment by JanAckermann** [comment]
@jvillafanez 
Don't worry. 

I tried it, and it looks good to me:
image

File: apps/files_external/js/settings.js [hunk 1]
```diff
@@ -1279,13 +1281,39 @@ MountConfigListView.prototype = _.extend({
...
 		});
 	},
 
+	/**
+	 * Recheck storage availability
+	 *
+	 * @param {jQuery} $tr storage row
+	 * @param {number} status storage status
+	 */
+	onRecheckStorageConfig: function($tr, status){
```
Comment (jvillafanez) [inline apps/files_external/js/settings.js:1300]: I think the "on" prefix is usually used when the function is directly bound to an event. In this case, it's the anonymous function the one bound, and this function is called by that anonymous function.
Changing the filename to "recheckStorageConfig" is enough to me.

Comment (JanAckermann) [reply apps/files_external/js/settings.js:1315]: Nope the function caller is recheckStorageConfig and onRecheckStorageConfig is a  callback,
so we can call it recheckStorageConfigCallBack aswell. 
I don't have strong feelings about it, its on you.

File: .htaccess [hunk 2]
```diff
@@ -94,3 +94,35 @@ Options -Indexes
 <IfModule pagespeed_module>
   ModPagespeed Off
 </IfModule>
+#### DO NOT CHANGE ANYTHING ABOVE THIS LINE ####
```
[the remark this answers is not in the document]
Comment (JanAckermann) [reply .htaccess:97]: whoops

File: apps/files_external/js/settings.js (hunk 1 above)
Comment (jvillafanez) [reply apps/files_external/js/settings.js:1311]: Ok, there is already a "recheckStorageConfig" method... leave it as it is.

**Review by IljaN — APPROVED** [review]

**Review by phil-davis — COMMENTED** [review]
Test changes LGTM and pass.

**Pull request merged**